src/main/java/org/folio/fqm/repository/EntityTypeRepository.java

```diff
@@ -47,6 +49,40 @@ public Optional<String> getDerivedTableName(UUID entityTypeId) {
       .fetchOptional(derivedTableNameField);
   }
 
+  public List<EntityTypeColumn> fetchNamesForSingleCheckbox(UUID entityTypeId) {
+    log.info("Getting derived table name for entity type ID: {}", entityTypeId);
+    String sourceViewName = getDerivedTableName(entityTypeId).get();
+    List<EntityTypeColumn> entityColumnsList = new ArrayList<>();
+
+    String requiredFieldName = "jsonb" + "->>'name'";
+    String abc =  "jsonb" + "->>'refId'";
+    String whereClauseName = "jsonb" + "->>'type'";
+    String customFieldType = "SINGLE_CHECKBOX";
+    ValueWithLabel valueWithLabelTrue = new ValueWithLabel().label("True").value("true");
+    ValueWithLabel valueWithLabelFalse = new ValueWithLabel().label("False").value("false");
+
+    Result<Record2<Object, Object>> result = jooqContext
+      .select(field(requiredFieldName), field(abc))
+      .from(sourceViewName)
+      .where(field(whereClauseName).eq(customFieldType))
```

Comment:
Rather than rebuilding these fields here every time this method is called, maybe store the `Field` objects as `static` fields on the class